source/solvers/fluid_dynamics_matrix_free.cc

```diff
@@ -2287,19 +2289,12 @@ FluidDynamicsMatrixFree<dim>::set_initial_condition_fd(
       this->set_nodal_values();
       this->present_solution.update_ghost_values();
 
-      // Create a pointer to the current viscosity model
-      std::shared_ptr<RheologicalModel> viscosity_model =
-        this->simulation_parameters.physical_properties_manager.get_rheology();
-
-      // Keep in memory the initial viscosity
-      const double viscosity_end = viscosity_model->get_kinematic_viscosity();
-
-      // Set it to the initial condition viscosity
-      viscosity_model->set_kinematic_viscosity(
-        this->simulation_parameters.initial_condition->kinematic_viscosity);
+      const double viscosity_end =
+        this->simulation_parameters.physical_properties_manager.get_rheology()
+          ->get_kinematic_viscosity();
 
-      // Set the kinematic viscosity in the system operator to be the
-      // temporary viscosity
+      // Set the kinematic viscosity in the system operator to be the temporary
+      // viscosity
       this->system_operator->set_kinematic_viscosity(
         this->simulation_parameters.physical_properties_manager
           .get_kinematic_viscosity_scale());
```

Comment:
Why do you need to do that here? You could just set the viscosity in the rheology model and you would not need to specify it no? I am not sure you need to specify it in the viscosity model thorugh here? There is only one physical property manager after all.

Reply:
This has been removed and updated